Frameworks/UIKit/UITextField.mm

```diff
@@ -129,13 +129,17 @@ - (void)setText:(NSString*)text {
 
         [_secureModeLock lock];
         if (_secureTextMode) {
-            _passwordBox.password = _text;
+            if (![_passwordBox.password isEqualToString:_text]) {
+                _passwordBox.password = _text;
+            }
         } else {
-            _textBox.text = _text;
-            // Ensure caret at end of field in case we programmatically
-            // gain focus (becomeFirstResponder) after the text is set:
-            _textBox.selectionStart = [_text length];
-            _textBox.selectionLength = 0;
+            if (![_textBox.text isEqualToString:_text]) {
+                _textBox.text = _text;
+                // Ensure caret at end of field in case we programmatically
+                // gain focus (becomeFirstResponder) after the text is set:
+                _textBox.selectionStart = [_text length];
+                _textBox.selectionLength = 0;
```

Comment:
This doesn't seem like it should fix the issue; when you delete a character it will still jump, because `_text != _textBox.text`?